src/criticality/CriticalitySearchBase.C

```diff
@@ -0,0 +1,126 @@
+  params.addRequiredParam<Real>(
+      "minimum",
+      "Minimum for values to search over; the root must occur at a value greater than the minimum");
+  params.addRequiredParam<Real>(
+      "maximum",
+      "Maximum for values to search over; the root must occur at a value smaller than the maximum");
+  params.addRangeCheckedParam<Real>(
+      "tolerance",
+      1e-3,
+      "tolerance > 0",
+      "Absolute tolerance to converge multiplication factor; be aware that if too few particles "
+      "are used, statistical noise may require many criticality calculations to converge.");
+  params.addParam<MooseEnum>(
+      "estimator", getEigenvalueEnum(), "Type of eigenvalue estimator to use");
+  params.addClassDescription(
+      "Base class for defining parameters used in a criticality search in OpenMC.");
+  params.registerBase("CriticalitySearch");
+  params.registerSystemAttributeName("CriticalitySearch");
+  params.addPrivateParam<OpenMCCellAverageProblem *>("_openmc_problem");
+  return params;
+}
+
+CriticalitySearchBase::CriticalitySearchBase(const InputParameters & parameters)
+  : MooseObject(parameters),
+    OpenMCBase(this, parameters),
+    _maximum(getParam<Real>("maximum")),
+    _minimum(getParam<Real>("minimum")),
+    _tolerance(getParam<Real>("tolerance")),
+    _estimator(getParam<MooseEnum>("estimator").getEnum<eigenvalue::EigenvalueEnum>())
+{
+  if (_minimum >= _maximum)
+    paramError("minimum",
+               "The 'minimum' value (" + std::to_string(_minimum) +
+                   ") must be less than the 'maximum' value (" + std::to_string(_maximum) + ").");
+
+  auto pp_params = _factory.getValidParams("Receiver");
+  _openmc_problem->addPostprocessor("Receiver", _pp_name, pp_params);
+}
+
+void
+CriticalitySearchBase::searchForCriticality()
+{
+  _console << "Running criticality search in OpenMC for " << quantity() << " in range "
+           << std::to_string(_minimum) << " - " << std::to_string(_maximum) << " " << units() << " "
+           << std::endl;
+
+  std::function<Real(Real)> func;
+  func = [&](Real x)
+  {
+    // update the OpenMC model with a new parameter
+    updateOpenMCModel(x);
+    _inputs.push_back(x);
+
+    // re-run the model
+    int err = openmc_run();
+    if (err)
+      mooseError(openmc_err_msg);
+
+    // fetch k and return the residual, k-1
+    Real k = kMean(_estimator);
+    Real k_std_dev = kStandardDeviation(_estimator);
+    _k_values.push_back(k);
+    _k_std_dev_values.push_back(k_std_dev);
+
+    if (_tolerance < 3 * k_std_dev)
+      mooseDoOnce(mooseWarning(
+          "The 'tolerance' for the criticality search (" + std::to_string(_tolerance) +
+          ") is smaller than 3-sigma standard deviation in k (" + std::to_string(3 * k_std_dev) +
+          "); you may have to run a lot of criticality search points to converge to this "
+          "tolerance. You may want to loosen 'tolerance' or increase the number of particles."));
```

Comment:
What if the reason this is true is because of poor statistics? Might be worth considering a comment that they can lower their `k_std_dev` by using more particles

Reply:
I'm not sure I understand what is being suggested - the warning message tells the user they could "increase the number of particles" as one possible fix